### DISCUSSION TITLE
I want to talk about a new feature that I will add

### DISCUSSION THREAD

**Discussion by thisisshadow** [Sketchware-Pro/Sketchware-Pro#1179] [Ideas]
**1. Redesign Build Dialog**

I want to renew the compile/build dialog, first we will change the build dialog to material 3 style, then we will renew the quiz layout, that is, for example, we will remove options such as X O and add questions that are completely with chances and at the bottom there will be a progress bar about the build status.

**2. Build Service**

In order not to mandatory the build dialog, we will create a separate service, it can be a foreground service, we will do the build operations in this service instead of the design activity and thus we will not mandatory this dialog and even while the application is being built in the background, the user can navigate in different applications and see the build status via notification.

**3. Build Cache** (I have no intention of making this feature, I just wanted to say that it would be nice if we could)

This feature may be impossible or overly challenging for sketchware, but I think it would be great if we could do it. Android Studio has a build cache, that is, it caches previously compiled code/files and compiles only the modified files in the next build, so the build time can be reduced to seconds.

**Comment by ilyassesalama** [comment]
Seems great, good luck.

**Reply by thisisshadow** [reply]
Thanks bro

**Reply by thisisshadow** [reply]
![IMG_20240630_210127](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/116355867/6c7e8655-bfd9-45a7-a1ec-3f3e138e5f24)

Bro build dialog has already been refreshed, so I won't be refreshing this dialog but maybe I could make a build service.

**Reply by ilyassesalama** [reply]
Yeah, why not. However, building the project in background requires taking many things into consideration to be make it as stable as possible.

**Reply by thisisshadow** [reply]
Bro, I will add an option to the mod settings, of course it will not be a mandatory option such as the "build in the background" option. Will continue in old style when closed

**Reply by thisisshadow** [reply]
Can be kept closed until stabilized
